src/plugins/intel_npu/tools/multi-provider-inference-tool/tools/multi_provider_blobs_comparison.py

```diff
@@ -0,0 +1,305 @@
+                    provider_to_cmp_name,
+                    provider_to_cmp_result_data["data"][ref_output_name],
+                )
+            )
+        else:
+            if (
+                "provider"
+                not in provider_to_cmp_result_data["not_found_data"][ref_output_name].keys()
+            ):
+                provider_to_cmp_result_data["not_found_data"][ref_output_name]["provider"] = []
+            provider_to_cmp_result_data["not_found_data"][ref_output_name]["provider"].append(
+                ref_provider_name
+            )
+
+    # Fill outputs which are missed in reference data but exist in provider_to_cmp
+    for output_to_cmp_name in provider_to_cmp_result_data["outputs"]:
+        if output_to_cmp_name not in ref_provider_result_data["data"].keys():
+            if (
+                "provider"
+                not in ref_provider_result_data["not_found_data"][output_to_cmp_name].keys()
+            ):
+                ref_provider_result_data["not_found_data"][output_to_cmp_name]["provider"] = []
+            ref_provider_result_data["not_found_data"][output_to_cmp_name]["provider"].append(
+                provider_to_cmp_name
+            )
+    return provider_to_cmp_result_data
+
+
+def multi_provider_result_comparator(ref_provider, providers_to_compare, model_path, case_num):
+    printer = TensorsInfoPrinter()
+    model_name = get_model_name(model_path)
+    result = {"providers": [], "data": {}, "status": [], "not_found_data": {}}
+
+    # gather result for a reference provider
+    ref_provider_data_root_dir = Path(ref_provider, str(case_num))
+    try:
+        # read data from provider affiliated directories
+        ref_provider_model_tensors_descr = printer.deserialize_output_tensor_descriptions(
+            ref_provider_data_root_dir, model_name
+        )
+
+        # fill LHS default values
+        result["data"][ref_provider] = initialize_provider_result_data(
+            ref_provider, ref_provider_model_tensors_descr, {"std_correlation": 1}
+        )
+        result["providers"].append(ref_provider)
+    except RuntimeError as ex:
+        result["status"].append(
+            f"Cannot find results for the reference provider: {ref_provider}, err: {ex}"
+        )
+        result["not_found_data"][ref_provider] = {}
+
+    # gather results for providers to compare
+    for p in providers_to_compare:
+        provider_data_root_dir = Path(p, str(case_num))
+        try:
+            # read data from provider affiliated directories
+            provider_model_tensors_descr = printer.deserialize_output_tensor_descriptions(
+                provider_data_root_dir, model_name
+            )
+
+            # fill default values
+            result["data"][p] = initialize_provider_result_data(
+                p, provider_model_tensors_descr, {"std_correlation": "NaN"}
+            )
+            result["providers"].append(p)
+        except Exception as ex:
```

Comment:
Maybe we could use consistent exception handling - either catch specific exceptions or use a base class consistently.
You are using `RuntimeError` in some places, `Exception` in others.